include/fastdds/dds/subscriber/DataReader.hpp

```diff
@@ -116,87 +136,86 @@ class RTPS_DllAPI DataReader
      * @param [out] info Pointer to a SampleInfo_t structure to store first untaken sample information.
      * @return RETCODE_OK if sample info was returned. RETCODE_NO_DATA if there is no sample to take.
      */
-    ReturnCode_t get_first_untaken_info(
+    RTPS_DllAPI ReturnCode_t get_first_untaken_info(
             fastrtps::SampleInfo_t* info);
 
     /**
      * Get associated GUID
      * @return Associated GUID
      */
-    const fastrtps::rtps::GUID_t& guid();
+    RTPS_DllAPI const fastrtps::rtps::GUID_t& guid();
 
-    fastrtps::rtps::InstanceHandle_t get_instance_handle() const;
+    RTPS_DllAPI fastrtps::rtps::InstanceHandle_t get_instance_handle() const;
 
     /**
      * Get topic data type
      * @return Topic data type
      */
     TypeSupport type();
 
+    /**
+     * Get TopicDescription
+     * @return TopicDescription
+     */
+    const TopicDescription* topic() const;
```

Comment:
```suggestion
    const TopicDescription* get_topic_description() const;
```

Reply:
I will use `get_topicdescription`, as this is the method name on the standard